packages/plugin-npm-cli/sources/commands/npm/login.ts

```diff
@@ -0,0 +1,162 @@
+import {Configuration, MessageName}                     from '@berry/core';
+import {PluginConfiguration, StreamReport}              from '@berry/core';
+import {npmConfigUtils, npmHttpUtils}                   from '@berry/plugin-npm';
+import inquirer                                         from 'inquirer';
+import {Readable, Writable}                             from 'stream';
+import {URL}                                            from 'url';
+
+const SUCCESS_MESSAGE = `Successfully logged in`;
+const ERROR_MESSAGE = `Invalid Authentication`;
+
+// eslint-disable-next-line arca/no-default-export
+export default (clipanion: any, pluginConfiguration: PluginConfiguration) => clipanion
+
+  .command(`npm login [-r,--registry REGISTRY]`)
+  .categorize(`Npm-related commands`)
+  .describe(`login to registry`)
+
+  .detail(`
+    This command will ask you for your registry username, email, password, and One-time password (when it applies).
+  `)
+
+  .example(
+    `Login to the default registry`,
+    `yarn npm login`,
+  )
+
+  .example(
+    `Login to a custom registry`,
+    `yarn npm login --registry http://localhost:4873`,
+  )
+
+  .action(async ({cwd, stdin, stdout, registry}: {cwd: string, stdin: Readable, stdout: Writable, registry: string}) => {
+    const configuration = await Configuration.find(cwd, pluginConfiguration);
+
+    // @ts-ignore
+    const prompt = inquirer.createPromptModule({
+      input: stdin,
+      output: stdout,
+    });
+
+    const report = await StreamReport.start({configuration, stdout}, async report => {
+      const registryUrl = registry ? new URL(registry).toString() : null;
+      const credentials = await getCredentials(prompt);
+      const url = `/-/user/org.couchdb.user:${encodeURIComponent(credentials.name)}`;
+      const requestOptions = {
+        configuration,
+        ident: null,
+        json: true,
+        forceAuth: false,
+        registryUrl,
+      };
+
+      // First we try to login only using the credentials (no otp)
+      try {
+        const response = await npmHttpUtils.put(url, credentials, requestOptions);
+
+        // If login succeeds here, go ahead and save the token to the home config, report and exit
+        // @ts-ignore
+        await setAuthToken(configuration, response.token);
```

Comment:
You're writing the auth token to the wrong registry, right? You're always writing it to the default one, even with `-r,--registry` (or `-s,--scope`).